lib/active_merchant/billing/gateways/firstdata_e4.rb

```diff
@@ -263,7 +264,7 @@ def commit(action, request, credit_card = nil)
 
         Response.new(successful?(response), message_from(response), response,
           :test => test?,
-          :authorization => response_authorization(action, response, credit_card),
+          :authorization => successful?(response) ? response_authorization(action, response, credit_card) : nil,
```

Comment:
That'll teach 'em! :wink:

I don't really care either way in this specific case, just want to be cautious in general about making anyone think we support their broken calling code. :fearful: